### ISSUE TITLE
Reed Solomon redundancy data

### ISSUE THREAD

**Issue by robertcb1** [DazDSP/EasyDRF#2]
Could you please add the "Reed Solomon" redundancy data option ( .RS1 .RS2 .RS3 .RS4) to EasyDRF similar to EasyPAL?

**Comment by DazDSP** [comment]
I have been working on adding the RS coding feature for some time, however there is a problem with the way the decoder works...

If the file header is missed (due to noise or interference), the file segments are not recombined - hence they are not available to decode by later stages.

For the RS decoding to work, all the available data needs to be assembled together and passed to the RS decoder routine - even if some of it is missing.

The EasyDRF/WinDRM code is based on the Dream source code, and it uses C Classes which are something I'm not yet familiar with - so correcting this problem is difficult.

Perhaps if there is someone reading this who is more familiar with C Classes and/or the Dream source code they could offer suggestions on how to solve this problem?

It appears that the Dream source code was written in such a way as to not process the data if the MOT file header is missing. This makes sense, as the filename and filesize information is needed to reassemble the file.

The filename and filesize data can be sent in other ways, but it will not be EasyPal compatible. The EasyPal RS protocol does not appear to be published anywhere.

The author of EasyPal - Eric, VK4AES passed away some years ago. I approached the Queensland Digital Group (QDG) for information regarding the EasyPal RS protocol, but they were unable or unwilling to help.

**Comment by robertcb1** [comment]
Contact Ties PA0MBO. Its software TXAMADRM and RXAMADRM - has implemented RS coding and is compatible with EasyPAL.

**Comment by DazDSP** [comment]
I contacted Ties, but his software was based on DIORAMA, and not Dream - so the source code was very different.

Instead of making EasyDRF compatible with EasyPal, the protocol has been modified to address the single points of failure and make the system more reliable.

In just the last two days a solution was found to reading the partial buffer data, and the erasure list processing was also added to make the RS decoding as powerful as EasyPal.

Only RS4 mode has been implemented so far, but the results are very good.

This new code will be tested, tidied up and then uploaded here as soon as possible.

![RS4](https://user-images.githubusercontent.com/48916381/132801782-9a5d2aaa-6d70-4d88-bed2-7a62c4d2aea2.png)
